src/main.rs

```diff
@@ -3,7 +3,14 @@ use std::{env, fs};
 
 fn main() -> Result<()> {
     let config = config()?;
-    let (input_device, output_device) = audio::devices()?;
+
+    let input_device = config
+        .audio
+        .as_ref()
```

Comment:
A little type shenanigans here. Using `config.audio` directly would destructure `config`, making it no longer usable, but I need it further on to create the pipeline. [`as_ref`](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_ref) lets me get a reference to the value inside the `Option<Audio>` there, avoiding the destructuring.